src/utils/server-utils.ts

```diff
@@ -270,3 +278,119 @@ export async function getLayersAvailableDates(): Promise<AvailableDates> {
+ * Format value from featureInfo response based on LabelType provided
+ *
+ * @return a formatted string
+ */
+function formatFeatureInfo(value: string, type: LabelType): string {
+  if (type === LabelType.Date) {
+    return `${moment(value).utc().format('MMMM Do YYYY, h:mm:ss')} UTC`;
+  }
+
+  return value;
+}
+
+/**
+ * Executes a getFeatureInfo request
+ *
+ * @return object of key: string - value: string with formatted values given label type.
+ */
+async function runFeatureInfoRequest(
+  url: string,
+  wmsParams: requestFeatureInfo,
+  layers: WMSLayerProps[],
+): Promise<{ [name: string]: string }> {
+  // Transform to snake case.
+  const toSnakeWmsParams = Object.entries(wmsParams).reduce(
+    (obj, item) => ({
+      ...obj,
+      [item[0].replace(
+        /[A-Z]/g,
+        letter => `_${letter.toLowerCase()}`,
+      )]: item[1],
+    }),
+    {},
+  );
+
+  const res = await fetch(formatUrl(`${url}/ows`, toSnakeWmsParams));
+  const resJson: GeoJSON.FeatureCollection = await res.json();
+
+  const parsedProps = resJson.features.map(feature => {
+    // Get fields from layer configuration.
+    const [layerId] = (feature?.id as string).split('.');
+
+    const featureInfoProps =
+      layers?.find(l => l.serverLayerName === layerId)?.featureInfoProps || {};
+
+    const searchProps = Object.keys(featureInfoProps);
+
+    const properties = feature.properties ?? {};
+
+    return Object.keys(properties)
+      .filter(k => searchProps.includes(k))
+      .reduce(
+        (obj, key) => ({
+          ...obj,
+          [featureInfoProps[key].label]: formatFeatureInfo(
+            properties[key],
+            featureInfoProps[key].type,
+          ),
+        }),
+        {},
+      );
+  });
+
+  return parsedProps.reduce((obj, item) => ({ ...obj, ...item }), {});
+}
+
+/**
+ * This function builds and runs the getFeatureInfo request given the parameters
+ *
+ * @return Promise with returned object from request
+ */
+function fetchFeatureInfo(
+  layers: WMSLayerProps[],
+  url: string,
+  params: FeatureInfoType,
+): Promise<{ [name: string]: string }> {
+  const requestLayers = layers.filter(l => l.baseUrl === url);
+  const layerNames = requestLayers.map(l => l.serverLayerName).join(',');
+
+  const requestParams = {
+    service: 'WMS',
+    request: 'getFeatureInfo',
+    version: '1.1.1',
+    exceptions: 'application/json',
+    infoFormat: 'application/json',
+    layers: layerNames,
+    srs: 'EPSG:4326',
+    queryLayers: layerNames,
+    featureCount: 1,
+    format: 'image/png',
+    styles: '',
+  };
+
+  const wmsParams: requestFeatureInfo = { ...params, ...requestParams };
+
+  return runFeatureInfoRequest(url, wmsParams, layers);
+}
+
+/**
+ * Collects all urls to create a getFeatureInfo request.
+ *
+ * @return Promise with returned object from request
+ */
+export async function makeFeatureInfoRequest(
+  layers: WMSLayerProps[],
+  params: FeatureInfoType,
+): Promise<{ [name: string]: string } | null> {
```

Comment:
Unless you're using an api without strong types, or or an untyped library we shouldn't make lose types like this. FOr `MapEvent` , it is fair to use `any` since the library doesn't give us a type, but from what I see we can make strong types using `GeoJSON.FeatureCollection`. If you can't see an easy way to create types for this let me know :)

Reply:
Geojson standard has a strong type, but `properties` field is a key, value container. It lets flexibility for including a wide range of possible keys and values. For PRISM and WFS, we extract the values from keys included within the layers.json file. I mean, we could add a type constraint for the response. However it would work only for a single layer. Since ADAM layers have different property schemas accross its layers

Reply:
I'm not diving into the details here, but would like to add that flexibility on the types of data returned from the WFS feature request is important. We're solving for one dataset now (ADAM tropical storms), but more will come down the road, and we won't know how information is structured for each layer until we get into the integration steps. So being able to manage various types within the configuration of the layer is preferred.